src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -209,13 +213,12 @@ private static bool IncludeDocument(Document document, string? clientName)
             ClientCapabilities clientCapabilities,
             CancellationToken cancellationToken)
         {
-            // Being asked about this document for the first time.  Or being asked again and we have different
-            // diagnostics.  Compute and report the current diagnostics info for this document.
-
-            // Razor has a separate option for determining if they should be in push or pull mode.
-            var diagnosticMode = document.IsRazorDocument()
-                ? InternalDiagnosticsOptions.RazorDiagnosticMode
-                : InternalDiagnosticsOptions.NormalDiagnosticMode;
+            var diagnosticMode = _serverKind switch
+            {
+                WellKnownLspServerKinds.LiveShareLspServer => InternalDiagnosticsOptions.LiveShareDiagnosticMode,
+                WellKnownLspServerKinds.RazorLspServer => InternalDiagnosticsOptions.RazorDiagnosticMode,
+                _ => InternalDiagnosticsOptions.NormalDiagnosticMode,
```

Comment:
btw, has razor run into any issues?  at this point i'd be ok just getting rid of the flag for it.  the flags primarily existed in teh beginning to allow us to easily back out if we ran into a problem.  but they're not intended to live forever (doesn't have to be in this PR).

Reply:
there have been bugs, but they are actively fixed and we aren't telling people to switch back to push because its clearly a subpar experience.